portafly/src/tests/components/shared/data-list/DataListTable.test.tsx

```diff
@@ -0,0 +1,304 @@
+import { fireEvent, within } from '@testing-library/react'
+import { sortable } from '@patternfly/react-table'
+
+jest.mock('components/shared/data-list/DataListContext')
+const useTableMock = useDataListTable as jest.Mock
+const usePaginationMock = useDataListPagination as jest.Mock
+const useFiltersMock = useDataListFilters as jest.Mock
+const useBulkActionsMock = useDataListBulkActions as jest.Mock
+
+const columns = [
+  {
+    categoryName: 'name',
+    title: 'Name',
+    transforms: [sortable]
+  },
+  {
+    categoryName: 'species',
+    title: 'Species / Race',
+    transforms: [sortable]
+  },
+  {
+    categoryName: 'nationality',
+    title: 'Nationality',
+    transforms: [sortable]
+  },
+  {
+    categoryName: 'state',
+    title: 'State',
+    transforms: [sortable]
+  }
+]
+
+const characters = [
+  {
+    id: 1,
+    name: 'Calanthe',
+    species: 'Human',
+    nationality: 'Cintra',
+    state: 'Dead'
+  },
+  {
+    id: 2,
+    name: 'Geralt',
+    species: 'Witcher',
+    nationality: '',
+    state: 'Alive'
+  },
+  {
+    id: 3,
+    name: 'Cirilla',
+    species: 'Human',
+    nationality: 'Cintra',
+    state: 'Alive'
+  },
+  {
+    id: 4,
+    name: 'Zoltan',
+    species: 'Dwarf',
+    nationality: 'Mahakam',
+    state: 'Alive'
+  }
+]
+
+const rows = characters.map((c) => ({
+  id: c.id,
+  cells: Object.keys(c).filter((k) => k !== 'id').map((k) => c[k]),
+  selected: false
+}))
+
+const isChecked = (c: HTMLElement) => (c as HTMLInputElement).checked
+
+const defaultTable = {
+  rows,
+  columns,
+  selectedRows: [],
+  sortBy: {},
+  setSortBy: jest.fn(),
+  selectOne: jest.fn(),
+  selectPage: jest.fn(),
+  selectAll: jest.fn()
+}
+const resetPagination = jest.fn()
+const defaultPagination = { startIdx: 0, endIdx: 5, resetPagination }
+const defaultFilters = { filters: {} }
+const defaultBulkActions = {}
+const resetMocks = () => {
+  useTableMock.mockReturnValue(defaultTable)
+  usePaginationMock.mockReturnValue(defaultPagination)
+  useFiltersMock.mockReturnValue(defaultFilters)
+  useBulkActionsMock.mockReturnValue(defaultBulkActions)
+}
+beforeAll(resetMocks)
+afterEach(resetMocks)
+
+const setup = () => {
+  const wrapper = render(<DataListTable />)
+  const getRow = (character) => wrapper.getByText(character.name).closest('tr') as HTMLElement
+  const bulkSelectorDropdown = within(wrapper.container.querySelector('#data-list-bulk-selector-dropdown') as HTMLElement)
+  const bulkActionsDropdown = within(wrapper.container.querySelector('#data-list-bulk-actions-dropdown') as HTMLElement)
+
+  return {
+    ...wrapper,
+    getRow,
+    bulkSelectorDropdown,
+    bulkActionsDropdown
+  }
+}
+
+it('should be able to sort columns', () => {
+  const { container } = setup()
+  const table = within(container.querySelector('table') as HTMLElement)
+  fireEvent.click(table.getByText(/Name/))
+
+  expect(defaultTable.setSortBy).toHaveBeenLastCalledWith(1, 'asc', true)
```

Comment:
I've recently done this for the ApplicationsTable: since the first number is the column index and it depends on whether the table is selectable or not, I think it's better to simply expect any number.
```suggestion
  expect(defaultTable.setSortBy).toHaveBeenLastCalledWith(expect.any(Number), 'asc', true)
```

Reply:
Sounds like a good enough assertion, yeah. Will give it a second thought, but if not, will do the suggestion